cms/admin/pageadmin.py

```diff
@@ -416,9 +417,15 @@ def change_view(self, request, object_id, form_url='', extra_context=None):
 
         response = super().change_view(
             request, object_id, form_url=form_url, extra_context=extra_context)
-        if tab_language and response.status_code == 302 and response._headers['location'][1] == request.path_info:
-            location = response._headers['location']
-            response._headers['location'] = (location[0], "%s?language=%s" % (location[1], tab_language))
+        if not DJANGO_3_2:
```

Comment:
There's a lot of duplicated code for the header change. We should just use a new variable for the header I think. Something like, this also makes it much clearer for anyone cleaning out support logic in the future:

```
# support < django 3.2
if not DJANGO_3_2:
    response_headers = "_headers"
# support >= django 3.2
else:
    response_headers = "headers"

if tab_language and response.status_code == 302 and response['response_headers']['location'][1] == request.path_info:
    location = response['response_headers']['location']
    response['response_headers']['location'] = (location[0], "%s?language=%s" % (location[1], tab_language))
```

I'm not sure if we created a reference to response.headers and then made a change to the value of our new variable would have changed the reference or created a new reference to the value. This may work as:

```
# support < django 3.2
if not DJANGO_3_2:
    response_headers = response._headers
# support >= django 3.2
else:
    response_headers = response.headers

if tab_language and response.status_code == 302 and response_headers['location'][1] == request.path_info:
    location = response['response_headers']['location']
    response_headers['location'] = (location[0], "%s?language=%s" % (location[1], tab_language))
```

The manual test for the above is that the following memory references are the same, if not the code above doesn't work and a setter would be required:

```
if not DJANGO_3_2:
    id(response_headers) == id(response._headers)
else:
    id(response_headers) == id(response.headers)
```